src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -7399,21 +7457,34 @@ subroutine summarize_timestep(domain)
                              realArgs=(/global_scalar_min, latMax_global, lonMax_global/))
 
          scalar_max = -1.0e20
+         offset_1d = huge(1)
          indexMax = -1
          kMax = -1
          latMax = 0.0
          lonMax = 0.0
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(max:scalar_max)
          do iEdge = 1, nEdgesSolve
          do k = 1, nVertLevels
-            if (u(k,iEdge) > scalar_max) then
-               scalar_max = u(k,iEdge)
-               indexMax = iEdge
-               kMax = k
-               latMax = latEdge(iEdge)
-               lonMax = lonEdge(iEdge)
+            scalar_max = max(scalar_max, u(k,iEdge))
+         end do
+         end do
+         !$acc end parallel
+
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(min:lin_indexMax)
+         do iEdge = 1, nEdgesSolve
+         do k = 1, nVertLevels
+            if (u(k,iEdge) == scalar_max) then
+               offset_1d = min(offset_1d, k + size(u,1)*(iEdge-1))
             end if
          end do
          end do
+         !$acc end parallel
+         kMax = modulo(offset_1d, size(u,1))
```

Comment:
I don't think the logic is completely correct if the min/max value is at `k = nVertLevels`. Instead, I think the logic should be:
```fortran
    do iEdge = 1, nEdgesSolve
    do k = 1, nVertLevels
       if (u(k,iEdge) == scalar_max) then
          offset_1d = min(offset_1d, (k-1) + size(u,1)*(iEdge-1))
       end if
    end do
    end do

    kMax = mod(offset_1d, size(u,1)) + 1
    indexMax = offset_1d / size(u,1) + 1
```

Reply:
I see what you mean. Update coming in soon.

E.g. if max is at `(k,iEdge) = (4,2)` and `(nVertLevels,nEdges) = (4,3)`, the current method would wind up with `kMax = 0` and `indexMax = 9`...

Reply:
Should be fixed as part of force-push [27dca1f](https://github.com/MPAS-Dev/MPAS-Model/commit/27dca1f1753d31d791471081aba358ae60065b2e) to [393492e](https://github.com/MPAS-Dev/MPAS-Model/commit/393492e813271638a1c9d48a76e0b010f26ea714). Let me know if not.